clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -4946,6 +4946,10 @@ void Clang::ConstructJob(Compilation &C, const JobAction &JA,
     if (HasFPGA)
       CmdArgs.push_back("-fsycl-disable-range-rounding");
 
+    // Pass -fintelfpga to both the host and device SYCL compilations if set.
+    if (Args.hasArg(options::OPT_fintelfpga))
+      CmdArgs.push_back("-fintelfpga");
```

Comment:
Use of `-fintelfpga` is equivalent to `-fsycl-targets=spir64_fpga` with some additional settings (deps generation, debug).  Should this setting be based on `-fintelfpga` on the command line or by virtue of the spir64_fpga triple?

Reply:
I like Mike's suggestion. Please verify but I think this is what you need  - llvm::Triple::SPIRSubArch_fpga, based on my memory of discussions for https://github.com/intel/llvm/pull/4824.

Reply:
> Use of `-fintelfpga` is equivalent to `-fsycl-targets=spir64_fpga` with some additional settings (deps generation, debug). Should this setting be based on `-fintelfpga` on the command line or by virtue of the spir64_fpga triple?

My understanding is that we need to allow _Bitint of size larger than 128 bits on both the host and the device when 
-fintelfpga is allowed.  Checking for the spir64_fpga triple will certainly handle it for the device, but for the host there is no difference in the triple when -fintelfpga is specified vs not.   By passing -fintelfpga to the host and device cc1 compilatons, I am able to check just this flag to enable it.

Reply:
@mdtoguchi, I am getting back to this one again since the downstream errors seem to have been resolved.

Could you elaborate a bit on how I can check if FPGA is enabled inside the FE sources?   Wouldn't I need to duplicate how `HasFPGA` is computed again over in SemaSYCL?